s3.go

```diff
@@ -0,0 +1,57 @@
+package veneur
+
+import (
+	"errors"
+	"io"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+)
+
+const S3Bucket = "stripe-veneur"
+
+const DefaultAWSRegion = "us-west-2"
+const AwsProfile = "veneur-s3-test"
+
+var svc s3iface.S3API
+
+var S3ClientUninitializedError = errors.New("s3 client has not been initialized")
+
+// credentials will be pull from environment variables
+// AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+
+func init() {
+	sess := session.New(&aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	_, err := sess.Config.Credentials.Get()
+	if err == nil {
+		svc = s3.New(sess)
+	}
+}
+
+func s3Post(hostname string, data io.ReadSeeker) error {
+	if svc == nil {
+		return S3ClientUninitializedError
+	}
+	params := &s3.PutObjectInput{
+		Bucket: aws.String(S3Bucket),
+		Key:    s3Path(hostname),
+		Body:   data,
+	}
+
+	_, err := svc.PutObject(params)
+	return err
+}
+
+func s3Path(hostname string) *string {
+	t := time.Now()
```

Comment:
random thought: with lots of things in the flush path referring to time.Now(), it'd be interesting to refactor out the time dependency so that we only retrieve time at the start of the flush (via a mockable function) and then pass that down to the other things that care about it.

anyway not saying we should do that, but it's something that came to mind while reading this piece of code. food for thought